DevWs for NIE-TECH CO., LTD. containing containing Min Smart Plug Dimmer

sky-nie opened the pull request.

Winniewen18 committed:
DevWs for NIE-TECH CO., LTD. containing containing Min Smart Plug Dimmer

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,672 @@
+            capability "Switch"
+            capability "Temperature Measurement"
+            capability "Switch Level"
+            capability "Light"
```
This capability is outdated

sky-nie replied:
@PKacprowiczS :
Do you mean that the code from line 54 to line 57 is out of date? Or just line 57? My current modification is to remove the 57th line.

PKacprowiczS replied:
Yeah, just `Light` capability.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,672 @@
+		details(["switch", "temperature", "refresh", "syncStatus", "sync", "firmwareVersion"])
+	}
+
+    simulator { }
```
This block isn't necessary.

sky-nie replied:
OK, I remove the line 117.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,672 @@
+            namespace: "sky-nie",
+            author: "winnie",
+       //     mnmn: "SmartThings",
+       //     vid:"generic-dimmer-3",
```
If you don't need `mnmn` and `vid` to be defined, just remove it.

sky-nie replied:
OK, I removed it.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,672 @@
+private initialize() {
+    def checkInterval = ((60 * 60 * 3) + (5 * 60))
+
+    def checkIntervalEvt = [name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"]]
```
In https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/69768/files#diff-ede2fdd21911fde9f7d3dd84c6f9f40dc678f9b3f53f81350d4f6a9c30230c5fR234 you wrote that this is _sleepy_ device, but here you've marked it as `offlinePingable`, which is definitely not true for sleepy devices. Aside from the fact that this option is reserved only for crucial devices (like home security), Dimmer Plug device doesn't seem to be a sleepy device.

So, to summarize, please remove `offlinePingable: "1"`, and...

sky-nie replied:
OK, I removed it.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,672 @@
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	logDebug "ping()"	
+}
```
...implement proper `ping()` method which will be able to bring device back from offline state. (In other words, which will send command to which device will response, and an event will be created)

sky-nie replied:
OK, I removed the line 236.

Winniewen18 committed:
delete dummy code for pull request.

sky-nie commented:
@PKacprowiczS :
I have updated the branch "devws-for-nie-tech-co-ltd-contai_49", You can review at any time convenient for you, thank you!

PKacprowiczS commented:
Answer me this question: is this really a sleepy device? Hard to believe that Dimmer Switch is actually sleepy.

Winniewen18 committed:
1.Uniform parameter organization;
2.restore codes about checkInterval except parameter offlinePingable.

sky-nie commented:
> Answer me this question: is this really a sleepy device? Hard to believe that Dimmer Switch is actually sleepy.

MP21ZD And  MP22ZD with this DTH, are devices designed with micro battery. They will not go to sleep.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,656 @@
+    }
+    refreshSyncStatus()
+    return []
+}
```
Since you've confirmed that it isn't a sleepy device, can you remove any misleading code/comments stating otherwise? Also include some commands in _ping()_ command.

sky-nie replied:
I delete code as you list, I'm not sure others.

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,656 @@
+        logDebug "${eventMap.descriptionText}"
+    }
+    return eventMap
+}
```
Implementing such method and using it once seems like an overkill. I guess you can just remove it.

sky-nie replied:
Yes，you are right ！

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,656 @@
+    else {
+        logTrace(desc)
+    }
+}
```
This will be checked by Device API, remove this method and just send an event where you're calling it.

sky-nie replied:
I replace it with sendEvent.

Winniewen18 committed:
change code as reviewer request.

Winniewen18 committed:
1.Syntax format compliance adjustment
2.delete dummy code

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,507 @@
+    cmds1.each {
+        actions << new physicalgraph.device.HubAction(it)
+    }
+    sendHubCommand(actions, 100)
```
I think that 
```
sendHubCommands(cmds, 500)
```
would be sufficient here, but please verify it (and change it, if that's true).

sky-nie replied:
The function "sendHubCommands" is a custom function and is only called once.  I just copied the function body to the place of call.

PKacprowiczS replied:
`sendHubCommand` isn't only a custom function - it is available for you right away. 
If you call it like this
```
def cmds = [List, of, commands]
sendHubCommand(cmds, delayBetweenCommands)
```
it'll be sufficient. You don't even have to call `physicalgraph.device.HubAction(cmd)`.

sky-nie replied:
OK, I have changed it as you said.

PKacprowiczS reviewed (changes requested):
I think that's the last thing from me. Please check it, and we can proceed further, I suppose.

Winniewen18 committed:
Syntax format compliance adjustment

PKacprowiczS reviewed (changes requested):
Just referring to one of my comments to get your attention https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/69768#discussion_r669750343

Winniewen18 committed:
1.Syntax format compliance adjustment
2.Adjust the preferences interface prompts for SmartTings App
3.Simplify the process of calling sendHubCommand

PKacprowiczS commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -140,17 +145,9 @@ def executeConfigureCmds() {
-    def actions = []
-    cmds?.each {
-        actions << new physicalgraph.device.HubAction(it)
+        def commands = [List, of, cmds]
```
What I wrote earlier https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/69768#discussion_r669750343 was an example. It meant that object `cmds` is literally a "List of commands". Not a list containing variables `List` (which, as a side note, is a type, not a variable), `of` (which isn't even declared*) and `commands`.
In your code, you'd just use 
```
sendHubCommand(cmds, 500)
```

\* \- which additionally proves me, that you don't test your code with a device (cause you'd get an exception). Keep in mind that is your responsibility to make sure that device works correctly under your code, and it's your job to maintain it later.

sky-nie replied:
Oh, A super low-level, stupid error...

Winniewen18 committed:
fix a bug about sendHubCommand

PKacprowiczS reviewed (dismissed):
@greens

greens commented on devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy:
```diff
@@ -0,0 +1,496 @@
+                cmds << secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: 0, sensorType: 1))
+            } else {
+                cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+            }
```
because of this code, your temperature report threshold will never be marked as synced

sky-nie replied:
oh, I checked the history version, the bug is happened for my sync mistake.

sky-nie replied:
@KKlimczukS @greens I have updated [the PR](https://github.com/sky-nie/SmartThingsPublic/tree/devws-for-nie-tech-co-ltd-contai_48). Can you review it at a convenient time?

Winniewen18 committed:
fix a bug about temperature report threshold sync.

Winniewen18 committed:
Syntax format compliance adjustment

Winniewen18 committed:
1. remove code about "Temperature Measurement" for release's product.
2. change "auto off interval"  and "auto on interval" 's range for release's product.

Winniewen18 committed:
add a fingerprint for a new device

greens commented:
Should be live in production next Tuesday.
